jobsAdmin/JobsAdmin.go

```diff
@@ -311,26 +311,11 @@ func (ja *jobsAdmin) JobMgrCleanUp(jobId common.JobID) {
 
 		// Delete the jobMgr from jobIDtoJobMgr map, so that next call will fail.
 		ja.DeleteJob(jobId)
-		jm.Log(pipeline.LogInfo, "Job deleted from jobMgr map")
-
-		// Call jm.Cancel to signal routines workdone.
-		// This will take care of any jobPartMgr release.
-		jm.Cancel()
-
-		// Cleanup the JobStatusMgr go routine.
-		jm.JobStatusMgrClean()
 
-		// Remove JobPartsMgr from jobPartMgr kv.
-		jm.JobPartsMgrsDelete()
-
-		// Transfer Thread Cleanup.
-		jm.TransferRoutineCleanup()
-
-		// Close chunk status logger.
-		jm.ChunkStatusLoggerCleanup()
-		jm.Log(pipeline.LogInfo, "JobMgrDone Exit, Closing the log")
+		jm.Log(pipeline.LogInfo, "Job deleted from jobMgr map")
 
-		jm.CloseLog()
+		// Rest of jobMgr related cleanup done by DeferredCleanupJobMgr function.
+		go jm.DeferredCleanupJobMgr()
```

Comment:
Don't make this a separate function, as it doesn't have any significance without JobMgrCleanUp() and that way we are encouraging wrong calling possibilities (someone calls DeferredCleanupJobMgr() w/o JobMgrCleanUp()) and a comment is very important why we are doing this, else it can easily confuse a reader.

So do this here only, like


// Now that we have removed the jobMgr from the map,  no new caller will find it and hence cannot start any
// new activity using the jobMgr. We cleanup the resources of the jobMgr in a deferred manner as a safety net
// to allow processing any messages that may be in transit.
// NOTE: This is not really required but we don't want to miss any in-transit messages as some of the TODOs in 
//            the code suggest.
// TODO: Make this inline after proper testing.
go func() {
    time.Sleep(60 * time.Second)
    ...
}()